src/extension/content-script/batteries/Mixcloud.ts

```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+
+import getOriginData from "../originData";
+import { findLightningAddressInText, setLightningData } from "./helpers";
+
+const urlMatcher = /^https:\/\/www.mixcloud.com\/([^/]+)(\/)?([^/]+)?(\/)?$/;
+
+function battery(): void {
+  const urlParts = document.location.pathname.split("/");
+  const username = urlParts[1];
```

Comment:
are we able to get those directly from the regex?

Reply:
Yes! The Mixcloud API links are the same as the original ones but with `api.mixcloud.com/username/showname` instead of `www.mixcloud.com/username/showname`